pkg/window/strategy/sliding/sliding_test.go

```diff
@@ -0,0 +1,154 @@
+package sliding
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/window"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSliding_AssignWindow(t *testing.T) {
+	loc, _ := time.LoadLocation("UTC")
+	baseTime := time.Unix(1651129201, 0).In(loc)
+
+	type fields struct {
+		Length          time.Duration
+		PeriodInSeconds int
+	}
+
+	tests := []struct {
+		name      string
+		fields    fields
+		eventTime time.Time
+		want      []*window.IntervalWindow
+	}{
+		{
+			name: "2in_1min",
+			fields: fields{
+				Length:          120 * time.Second,
+				PeriodInSeconds: 60,
+			},
+			eventTime: baseTime,
+			want: []*window.IntervalWindow{
+				{
+					Start: time.Unix(1651129200, 0).In(loc),
+					End:   time.Unix(1651129320, 0).In(loc),
+				},
+				{
+					Start: time.Unix(1651129200-60, 0).In(loc),
+					End:   time.Unix(1651129320-60, 0).In(loc),
+				},
+			},
+		},
+		{
+			name: "2min-as-sec_1min",
+			fields: fields{
+				Length:          2 * time.Minute,
+				PeriodInSeconds: 60,
+			},
+			eventTime: baseTime,
+			want: []*window.IntervalWindow{
+				{
+					Start: time.Unix(1651129200, 0).In(loc),
+					End:   time.Unix(1651129200+120, 0).In(loc),
+				},
+				{
+					Start: time.Unix(1651129200-60, 0).In(loc),
+					End:   time.Unix(1651129200+120-60, 0).In(loc),
+				},
+			},
+		},
+		{
+			name: "4min_2min",
+			fields: fields{
+				Length:          4 * time.Minute,
+				PeriodInSeconds: 120,
+			},
+			eventTime: baseTime,
+			want: []*window.IntervalWindow{
+				{
+					Start: time.Unix(1651129200, 0).In(loc),
+					End:   time.Unix(1651129200+240, 0).In(loc),
+				},
+				{
+					Start: time.Unix(1651129200-120, 0).In(loc),
+					End:   time.Unix(1651129200+240-120, 0).In(loc),
+				},
+			},
+		},
+		{
+			name: "4min_4min",
+			fields: fields{
+				Length:          4 * time.Minute,
+				PeriodInSeconds: 240,
+			},
+			eventTime: baseTime,
+			want: []*window.IntervalWindow{
+				{
+					Start: time.Unix(1651129200, 0).In(loc),
+					End:   time.Unix(1651129200+240, 0).In(loc),
+				},
+			},
+		},
+		{
+			name: "4min_5min",
```

Comment:
What will happen with this kind of cases? Shall we treat it as invalid requirements?

Similarly, `4min_3min`.

Reply:
`4min_3min` is a valid input. if the period is greater than the length, then we can treat it as invalid for the time being.